### DISCUSSION TITLE
How do we define sheets functions, how to iterate arrays?

### DISCUSSION THREAD

**Discussion by c6p** [ananthakumaran/paisa#188] [Q&A]
I understand that I can `years_to_double(rate) = 72 / rate` to define a function. But I have trouble understanding doing anything more complex.

For example a query returns a `Posting[]`. Though, how do I iterate an array in sheets? I tried to write javascript, didn't work.
Is it possible for user to define javascript functions without recompile, and use them?

**Comment by ananthakumaran** [comment]
can you share your use cases? My motivation with the sheet is building something like https://soulver.app. It's still very new and experimental, but I don't think adding arrays, etc would help. If you tell me about your use case and if it can be solved by functions, that can be done.

**Comment by c6p** [comment]
I want to calculate gains of alternative investment paths. What if scenarios, i.e. calculate interest of commodities bought, sold at different dates. If I can define js functions somewhere without a recompile, current sheets functionality is enough for me.
